include/dirac_quda.h

```diff
@@ -1556,6 +1568,8 @@ namespace quda {
       @param[in] stream Which stream to run the prefetch in (default 0)
     */
     virtual void prefetch(QudaFieldLocation mem_space, qudaStream_t stream = device::get_default_stream()) const;
+    
+    void SmearOp(ColorSpinorField &out, const ColorSpinorField &in, const double &a, const double &b, const int &t0,  const QudaParity parity) const;    
```

Comment:
done.

Reply:
Thanks, but there's still some missing `in`/`out` attributes and the ones for `in` and `out` should be `@param[out] out` as opposed to `@param out[out]`

Reply:
Why do we have a `b (ignored)` parameter?

Reply:
Why do we pass doubles as const ref instead of by value?

Reply:
In fact, both `a` and `b` parameters do nothing in `Dirac(Improved)Staggered::SmearOP()`'s, since we pull them out as an overall factor. (For clarity, we omit their names in the definitions of these functions.)
However, we keep them in the input structure to accommodate the future addition of `GaugeLaplace::SmearOp()`.
 (See https://github.com/lattice/quda/blob/feature/quark-smearing/include/dirac_quda.h#L1959. I suppose Dean worked on it before us.)
If you want to remove them for now, that is okay with us.

Reply:
Regarding calling by reference, again, we follow the structure of `GaugeLaplace::SmearOp()` that started before this work. I don't think we have any preference for it.

Reply:
Okay. Alexei and I discussed it, and we decided to change these `a`,`b`,`t0` parameters to be called by value and fixed doxygen description for `a`. Thank you for bringing it up.
https://github.com/lattice/quda/commit/8bdf739483e267317bc4e7af7117b202907ecd35
https://github.com/lattice/quda/commit/ff3af36ad06315d7f9dea719009a3078fcbe85ce